src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -2799,6 +2907,20 @@ public override SyntaxNode WithStatements(SyntaxNode declaration, IEnumerable<Sy
             }
         }
 
+        private SyntaxNode FixDeclarationWithStatements(SyntaxNode declaration)
+        {
+            if (HasBody(declaration))
+            {
+                var mods = GetModifiers(declaration);
+                if (mods.IsAbstract)
+                {
+                    return WithModifiers(declaration, mods.WithIsAbstract(false).WithIsVirtual(true));
```

Comment:
I am saying that i disagree with the design.  I don't like the idea of just overriding explicit options the user asked for.  The purpose of this API for me when i originally wrote it was to serve as an agnostic way of generating C# and VB.  Making unrelated changes when the user asks for a specific change was not part of any design i can remember discussing.  

If i have an method with no body, and i make it virtual, then that does not mean that i now want it to have a body.  MAS type scenarios are a clear case where that sort of behavior is not desirable.  

Now, again, if this is something optional that the client can opt-in/out of, i'd be much more ok with this.  But i do not like unilateral action of this API to override the explicit options i passed in.